### ISSUE TITLE
Smooth & colored indent-guide

### ISSUE THREAD

**Issue by zim0369** [helix-editor/helix#4010]
Indent guides should look smoother and highlighted depending on the currently focused paragraph.

**Comment by zim0369** [comment]
The default indent-guides look good enough. Can they be replaced with a little thicker line?
```
character = "▏"
```
This isn't too thick. It looks decent.

Other than that, just highlighting the indent-guide depending on the scope.

**Comment by the-mikedavis** [comment]
I prefer the default character; I think it looks better to be centered rather than left-aligned.

What were you thinking for highlighting? I remember a chat in the Matrix room about using rainbow colors like in #2857

**Comment by s0LA1337** [comment]
> What were you thinking for highlighting? I remember a chat in the Matrix room about using rainbow colors like in #2857

I believe something like i have in my neovim config:
![image](https://user-images.githubusercontent.com/13474089/193369661-1d418fc0-822e-43f4-8d3a-770236194426.png)

**Comment by s0LA1337** [comment]
Well since i actually also wanted that feature, i figured to quickly implement this.

**Comment by musjj** [comment]
Rainbow guides are great, but based on OP's wording I don't think that's what they were asking. I think OP is asking for an option to highlight the indent guide of the paragraph that the cursor is currently on.

An example screenshot:

![](https://user-images.githubusercontent.com/72612857/273521224-6a18a9e4-1389-43d6-8458-ad01b536f8e0.png)

**Comment by evccyr** [comment]
> An example screenshot:
> 
> ![](https://user-images.githubusercontent.com/72612857/273521224-6a18a9e4-1389-43d6-8458-ad01b536f8e0.png)

Yes, this is what I meant. Personally I think that rainbow indents look cool but these are simple, subtle and more practical.

**Comment by carlosvigil** [comment]
I see 5 options:
0. off
1. on (the current guides)
2. active/current highlight
3. rainbow
4. rainbow with current highlight

**Comment by palango** [comment]
I think those steps are required to highlight the current scope using the indent guides. 

1. For each selection of a document, get the parent node via treesitter. Should be similar to 
https://github.com/helix-editor/helix/blob/cad0209e202a7513105639daa36c82578db4032d/helix-core/src/object.rs#L4
1. Pass the range information of the parent node to `render_text`.
2. Map the selection ranges to lines in the rendering code.
1. Pass those infos to `draw_indent_guides` and choose a different style there if inside the range and at the parents indention level.

I'd be happy to give this a try, but steps 2-4 are a bit unclear to me. Any hints are appreciated.
